pygeo/pyBlock.py

```diff
@@ -858,7 +855,31 @@ def projectPoints(self, x0, checkErrors, embTol, eps, nIter):
         v0 = 0.0
         w0 = 0.0
 
+        # If we are only interested in interior points, we skip projecting exterior points to save time.
+        # We identify exterior points by checking if they are outside the convex hull of the control points.
+        # A point can be inside the convex hull but still outside the FFD volume(s).
+        # In this case, the point is identified as an exterior point by the projection, which is more costly.
+        if interiorOnly:
+            # Compute the convex hull of all control points
+            hull = ConvexHull(self.coef)
+
+            # ConvexHull.equations describes the planes that define the faces of the convex hull
+            # Extract the normal vector and offset for each plane
+            hullNormals = hull.equations[:, :-1]
+            hullOffsets = hull.equations[:, -1]
+
+            # The normals point outside the convex hull, so a point is inside the convex hull if the distance in the
+            # normal direction from the point to every plane defining the convex hull is negative.
+            # This is computed in a vectorized manner below.
+            distanceToPlanes = np.dot(x0, hullNormals.T) + hullOffsets
```

Comment:
I am having a hard time getting the signs to make sense here, is the offset sign flipped?

Reply:
For each plane, the offset is the distance (in the normal direction) from the origin to plane. The dot product by itself gives us the distance from a point to a plane that intersects the origin and has the same normal as the actual plane. Adding the offset gives us the distance to the actual plane.

Reply:
After reading the Qhull documentation a bit more and printing out the offsets for a simple case, I realized you are correct. The offset is negative in the normal direction. I added a note about this in the comments.